src/nwbinspector/checks/time_series.py

```diff
@@ -88,6 +88,21 @@ def check_timestamps_ascending(time_series: TimeSeries, nelems=200):
         return InspectorMessage(f"{time_series.name} timestamps are not ascending.")
 
 
+@register_check(importance=Importance.BEST_PRACTICE_VIOLATION, neurodata_type=TimeSeries)
+def check_timestamp_of_the_first_sample_is_not_negative(time_series: TimeSeries):
+    """
+    Check that the timestamp of the first sample is not negative.
+
+    Best Practice: :ref:`best_practice_timestamp_of_the_first_sample`
+    """
+
+    first_timestamp = time_series.starting_time if time_series.starting_time is not None else time_series.timestamps[0]
```

Comment:
Just mentally checking, is it 100% guaranteed among all the `TimeSeries` children that they have either `starting_time` or `timestamps` set to `None`? Are there any other special time references, including and not limited to `starting_frame` in `ImageSeries` (with multiple external files)? Is it also true that `starting_time` is always set `None` as an attribute and is never missing (i.e., `hasattr(time_series, "starting_time") == False`?

Reply:
Recommend utilizing `pynwb.testing.mock` to test it out on various subtypes just to be sure (doesn't need to be in testing suite, just want to be confident we're not missing something)

Reply:
I'm pretty sure this holds across subtypes

Reply:
also: [FeatureExtraction](https://pynwb.readthedocs.io/en/stable/pynwb.ecephys.html#pynwb.ecephys.FeatureExtraction), [EventDetection](https://pynwb.readthedocs.io/en/stable/pynwb.ecephys.html#pynwb.ecephys.EventDetection) both have a `times` attribute.

Reply:
Oh, good find! Perhaps a separate check for those objects too then?

Reply:
Though perhaps in a separate PR; if those data types are deemed important enough to need it. I've never actually seen them used before.

This PR is already enough for the major cases

Reply:
me neither, just checked `pynwb` API documentation for time attributes.